src/platform/monitoring/Datadog/useBrowserMonitoring.js

```
@@ -0,0 +1,166 @@
+ * @property {boolean} forwardErrorsToLogs - Whether to forward errors to logs
+ * @property {Array<string>} forwardConsoleLogs - Console log levels to forward
+ * @property {Array<string>} forwardReports - Reports to forward
+ * @property {number} sessionSampleRate - Sample rate for sessions
+ * @property {string} service - Datadog service name
+ * @property {string} site - Datadog site (default: 'ddog-gov.com')
+ * @property {number} telemetrySampleRate - Sample rate for telemetry
+ * @property {string} version - Datadog version
+ */
+/**
+ * Initialize Datadog RUM directly, if not using a feature flag
+ * @param {CustomLogSettings} customLogSettings
+ */
+const initializeBrowserLogging = (
+  customLogSettings,
+  checkInit = canInitDatadog,
+) => {
+  // Prevent LOGS from re-initializing the SDK OR running on local/CI
+  // environments.
+  if (checkInit()) {
+    datadogLogs.init({
+      // Required settings
+      // applicationId: '{UUID}',
+      // clientToken: 'pubb{ID}',
+      // service: 'benefits-{app name}',
+      // version: '1.0.0',
+
+      // default settings
+      site: 'ddog-gov.com',
+      // see src/site/constants/vsp-environments.js for defaults
+      env: environment.vspEnvironment(), // 'production'
+      forwardConsoleLogs: ['error'],
+      forwardErrorsToLogs: true,
+      forwardReports: [],
+      silentMultipleInit: true, // silently ignore multiple inits
+      telemetrySampleRate: 100,
+      ...(customLogSettings || {}),
+    });
+  }
+};
+
+// Initialize Datadog RUM behind feature flag
+const useBrowserMonitoring = ({ toggleName = '', ...settings } = {}) => {
+  const {
+    TOGGLE_NAMES,
+    useToggleValue,
+    useToggleLoadingValue,
+  } = useFeatureToggle();
+
+  const isLoadingFeatureFlags = useToggleLoadingValue();
+  /* eslint-disable react-hooks/rules-of-hooks */
+  const isBrowserMonitoringEnabled = useToggleValue(TOGGLE_NAMES[toggleName]);
+  /* eslint-enable react-hooks/rules-of-hooks */
+
+  useEffect(
```

Any concerns over the useEffect callback potentially running multiple times unexpectedly? Is there ever a chance that the datadog init code runs more than once?

---

Datadog introduced a new setting `silentMultipleInit` that has been set to `true` in this code.
- [RUM settings](https://docs.datadoghq.com/real_user_monitoring/browser/setup/client/?tab=rum#initialization-parameters) (Initialization fails silently if the RUM Browser SDK is already initialized on the page.)
- [Log settings](https://docs.datadoghq.com/logs/log_collection/javascript/?tab=us#configuration) (Prevent logging errors while having multiple init.)

So, I don't think it'll be an issue